### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by orangemug** [maputnik/editor#337]
Maputnik needs better documentation. Over the course of the year there have been a few issues raised and comments in issues with people confused what Maputnik actually does. I don't think it's that clear from the project what it does and what it's aims are.

How about we create a documentation repository (maputnik/docs) and start building some good documentation.

My initial thoughts of what would be good to cover

 - The goal of Maputnik (an editor for the MapboxGL style spec)
 - Worked through use cases
   - Base map
   - Data vis
 - An explanation of data sources
 - Common issues when using Maputnik

So basically it'd be a guide to both using Maputnik and to some extent working with the _MapboxGl style spec_. Although if you have better ideas feel free to run with them

Would anybody be interested taking on the challenge? 

I would prefer someone with prior experience, I'm going to nominate @gregorywolanski @pathmapper @keum. I realise this is quite a lot of work, so there's no pressure intended from me.

**Comment by gregorywolanski** [comment]
This issue appears in the answers to question “What annoys you about Maputnik?” in the survey (original spelling):

- The lack of instructions about style properties.

- Hard to connect local datasource. 

- Hard to know what to start. 

- No toturial from raw data to maputnik to map tile.

- Missing docs. about eg. labelling.

- Lack of/unclear documentation.

- We do a lot of re-engeneering to get the detail bits out of a style.

- Some parts of the documentation and video tutorials are outdated and not replicable.

- I cant make the labeling work.

- lack of documentation to configure it from a developer point of view

**Comment by pathmapper** [comment]
Just a note that we may want to document https://github.com/maputnik/editor/issues/301.

**Comment by pathmapper** [comment]
- heatmap tutorial (#540)
